lib/output/cassandra.go

```diff
@@ -382,6 +342,50 @@ func (c *cassandraWriter) writeBatch(session *gocql.Session, msg types.Message)
 	return nil
 }
 
+func (c *cassandraWriter) parseArgs(msg types.Message, index int) ([]interface{}, error) {
+	// If we've been given the "args" field, extract values from there.
+	if len(c.conf.Args) > 0 {
+		var args []*field.Expression
+		for i, v := range c.conf.Args {
+			expr, err := bloblang.NewField(v)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse arg %v expression: %v", i, err)
+			}
+			args = append(args, expr)
+		}
+
+		values := make([]interface{}, 0, len(args))
+		for _, arg := range args {
+			values = append(values, stringValue(arg.String(0, msg)))
+		}
+
+		return values, nil
+	}
+
+	argsMapping, err := bloblang.NewMapping("", c.conf.ArgsMapping)
```

Comment:
It's possible that this parsing is actually quite expensive so we probably ought to do it only once in the constructor, and the same goes for the args fields. It's not as tidy but it also means we'd capture mapping errors earlier.